src/main/java/com/fakng/fakngagrgtr/parser/impl/google/GoogleParser.java

```diff
@@ -0,0 +1,86 @@
+package com.fakng.fakngagrgtr.parser.impl.google;
+
+import com.fakng.fakngagrgtr.entity.Vacancy;
+import com.fakng.fakngagrgtr.parser.ApiParser;
+import com.fakng.fakngagrgtr.parser.impl.google.dto.ResponseDTO;
+import com.fakng.fakngagrgtr.parser.impl.google.dto.VacancyDTO;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class GoogleParser extends ApiParser {
+
+    public GoogleParser(@Autowired WebClient webClient, @Value("${url.google}") String url) {
+        super(webClient);
+        this.url = url;
+    }
+
+    @Override
+    protected List<Vacancy> process() throws Exception {
+        ResponseDTO firstPage = getPage(1);
+        List<VacancyDTO> googleVacancies = new ArrayList<>(firstPage.getJobs());
+        for (int i = 2; i <= firstPage.getCount() / firstPage.getPageSize() + 1; i += 1) {
+            System.out.println("Page " + i);
+            ResponseDTO page = getPage(i);
+            googleVacancies.addAll(page.getJobs());
+        }
+        return googleVacancies.stream().map(this::createVacancy).toList();
```

Comment:
You can map those on every loop iteration to not hold all vacancies in mem

Reply:
Done.